packages/grid/test/drag-and-drop.common.js

```diff
@@ -1073,5 +1068,19 @@ describe('drag and drop', () => {
       fireDragOver(grid.__getViewportRows()[0], 'above');
       expect(grid.$.table.scrollTop).to.be.within(scrollTop - 200, scrollTop - 100);
     });
+
+    it('should add/remove drag-source- part when scrolling', () => {
+      grid.rowsDraggable = true;
+      grid.selectItem(grid.items[0]);
+      const renderedBufferCount = grid.$.items.childElementCount;
+      fireDragStart();
+
+      grid.scrollToIndex(renderedBufferCount);
+      expect(getFirstCell(grid).getAttribute('part')).to.not.contain('drag-source-row-cell');
+
+      grid.scrollToIndex(0);
+      expect(getFirstCell(grid).getAttribute('part')).to.contain('drag-source-row-cell');
+      grid.rowsDraggable = false;
+    });
```

Comment:
Overall the new test looks good, but using `getFirstCell` after scrolling to an index other than 0 wouldn't guarantee that we're checking the correct `<td>` element.

I'd suggest the following changes:
```js
it('should add/remove drag-source- part when scrolling', async () => {
  grid.rowsDraggable = true;
  grid.selectItem(grid.items[0]);
  fireDragStart();

  // Make sure the cell with drag-source-row-cell part is rendered
  await nextFrame();
  expect(getFirstCell(grid).getAttribute('part')).to.contain('drag-source-row-cell');

  // Scroll down so that the drag source cell leaves the viewport
  grid.scrollToIndex(50);
  // Expect no cells with drag-source-row-cell part in the DOM
  expect(grid.$.items.querySelector('tr:not([hidden]) [part~="drag-source-row-cell"]')).to.be.null;

  // Scroll back to the top
  grid.scrollToIndex(0);
  // Expect the first cell to have drag-source-row-cell part again
  expect(getFirstCell(grid).getAttribute('part')).to.contain('drag-source-row-cell');
});
```